File: win-prog/ProgWinVB/Brushes and Pens/LineCaps/LineCaps.vb
'-----------------------------------------
' LineCaps.vb (c) 2002 by Charles Petzold
'-----------------------------------------
Imports System
Imports System.Drawing
Imports System.Drawing.Drawing2D
Imports System.Windows.Forms

Class LineCaps
    Inherits PrintableForm

    Shared Shadows Sub Main()
        Application.Run(New LineCaps())
    End Sub

    Sub New()
        Text = "Line Caps"
    End Sub

    Protected Overrides Sub DoPage(ByVal grfx As Graphics, _
            ByVal clr As Color, ByVal cx As Integer, ByVal cy As Integer)
        Dim pnWide As New Pen(Color.Gray, Font.Height)
        Dim pnNarrow As New Pen(clr)
        Dim br As New SolidBrush(clr)
        Dim lc As LineCap

        For Each lc In System.Enum.GetValues(GetType(LineCap))
            grfx.DrawString(lc.ToString(), Font, br, _
                            Font.Height, Font.Height \ 2)
            pnWide.StartCap = lc
            pnWide.EndCap = lc
            grfx.DrawLine(pnWide, 2 * cx \ 4, Font.Height, _
                                  3 * cx \ 4, Font.Height)
            grfx.DrawLine(pnNarrow, 2 * cx \ 4, Font.Height, _
                                    3 * cx \ 4, Font.Height)
            grfx.TranslateTransform(0, 2 * Font.Height)
        Next lc
    End Sub
End Class

File: win-prog/ProgWinCS/Brushes and Pens/LineCaps/LineCaps.cs
//---------------------------------------
// LineCaps.cs © 2001 by Charles Petzold
//---------------------------------------
using System;
using System.Drawing;
using System.Drawing.Drawing2D;
using System.Windows.Forms;

class LineCaps: PrintableForm
{
     public new static void Main()
     {
          Application.Run(new LineCaps());
     }
     public LineCaps()
     {
          Text = "Line Caps";
     }
     protected override void DoPage(Graphics grfx, Color clr, int cx, int cy)
     {
          Pen   penWide   = new Pen(Color.Gray, Font.Height);
          Pen   penNarrow = new Pen(clr);
          Brush brush     = new SolidBrush(clr);

          foreach (LineCap lc in Enum.GetValues(typeof(LineCap)))
          {
               grfx.DrawString(lc.ToString(), Font, brush, 
                               Font.Height, Font.Height / 2);

               penWide.StartCap = lc;
               penWide.EndCap   = lc;

               grfx.DrawLine(penWide, 2 * cx / 4, Font.Height, 
                                      3 * cx / 4, Font.Height); 

               grfx.DrawLine(penNarrow, 2 * cx / 4, Font.Height, 
                                        3 * cx / 4, Font.Height); 

               grfx.TranslateTransform(0, 2 * Font.Height);
          }
     }
}
